### DISCUSSION TITLE
Is there a way to properly text search in a table?

### DISCUSSION THREAD

**Discussion by vigneshvelu** [Budibase/budibase#1934] [Help] [answered]
I am using a textfield component to search. I added a filter to the data provider which contains the table. In the filter, I mapped to 'starts with' and binded the text field. This feels very limited. I had to type the exact thing to get the row. Can only search text at the start. Is there a better way to do this? 

<img width="270" alt="Screenshot 2021-07-06 at 5 54 41 PM" src="https://user-images.githubusercontent.com/24852619/124599390-3ec34a00-de83-11eb-9e0a-ad9f015a8e26.png">

**Comment by mike12345567** [comment, marked as answer]
Hi @vigneshvelu - currently this is a restriction of CouchDB/Lucene searching, it does not allow very dynamic searching that you might get from something like Elasticsearch due to the performance considerations of creating such a complex index.

This is something we'll be looking at in the future, it will also be possible to do everything you can with an internal table against an SQL table in the future, which will allow much more complex search syntax (via queries with `LIKE` syntax).

**Reply by vigneshvelu** [reply]
Thank you for the answer @mike12345567
